src/3d/qgs3daxissettings.h

```diff
@@ -53,10 +64,25 @@ class _3D_EXPORT Qgs3DAxisSettings
     //! Writes settings to a DOM \a element
     void writeXml( QDomElement &element, const QgsReadWriteContext &context ) const;
 
+    //! Returns the default axis viewport size in pixel
+    int defaultViewportSize() const { return mDefaultViewportSize;}
+    //! Sets the defaultl axis viewport size in pixel
+    void setDefaultViewportSize( int size ) { mDefaultViewportSize = size; }
+
+    //! Returns the minimal axis viewport ratio (see Qt3DRender::QViewport::normalizedRect())
+    double minViewportRatio() const { return mMinViewportRatio;}
+    //! Sets the minimal axis viewport ratio between 0-1
+    void setMinViewportRatio( double ratio ) { mMinViewportRatio = ratio; }
+
+    //! Returns the maximal axis viewport ratio (see Qt3DRender::QViewport::normalizedRect())
+    double maxViewportRatio() const { return mMaxViewportRatio;}
+    //! Sets the maximal axis viewport ratio between 0-1
+    void setMaxViewportRatio( double ratio ) { mMaxViewportRatio = ratio; }
```

Comment:
What happens if  ratio > 1 or ratio < 0?

Reply:
and if mMinViewportRatio > mMaxViewportRatio

Reply:
well done!